### ISSUE TITLE
Restoring checkpoint and total loss

### ISSUE THREAD

**Issue by MI-LA01** [hunglc007/tensorflow-yolov4-tflite#7]
Hello, 
I trained from scratch a model w/ a custom dataset and ended with this : 
```
giou_loss: 5.81   conf_loss: 3.66   prob_loss: 1.10   total_loss: 10.57
```
When restoring checkpoints for training, assuming training data are the same, the conf_loss increase a lot after Step2

```
=> STEP    1   lr: 0.001000   giou_loss: 5.81   conf_loss: 3.66   prob_loss: 1.10   total_loss: 10.57
=> STEP    2   lr: 0.000000   giou_loss: 13.02   conf_loss: 75.97   prob_loss: 10.71   total_loss: 99.71
=> STEP    3   lr: 0.000000   giou_loss: 11.97   conf_loss: 75.60   prob_loss: 10.89   total_loss: 98.46
=> STEP    4   lr: 0.000000   giou_loss: 16.61   conf_loss: 96.86   prob_loss: 14.58   total_loss: 128.05
=> STEP    5   lr: 0.000000   giou_loss: 23.43   conf_loss: 93.97   prob_loss: 16.05   total_loss: 133.45
=> STEP    6   lr: 0.000000   giou_loss: 12.04   conf_loss: 80.76   prob_loss: 10.66   total_loss: 103.46
=> STEP    7   lr: 0.000000   giou_loss: 22.63   conf_loss: 97.42   prob_loss: 16.15   total_loss: 136.20
=> STEP    8   lr: 0.000000   giou_loss: 19.34   conf_loss: 93.15   prob_loss: 12.77   total_loss: 125.27
=> STEP    9   lr: 0.000000   giou_loss: 16.54   conf_loss: 77.26   prob_loss: 15.48   total_loss: 109.28
=> STEP   10   lr: 0.000000   giou_loss: 19.21   conf_loss: 89.43   prob_loss: 17.07   total_loss: 125.71
=> STEP   11   lr: 0.000000   giou_loss: 13.00   conf_loss: 81.75   prob_loss: 9.75   total_loss: 104.50
```

I tried in the config.py to reset the ```FISRT_STAGE_EPOCHS``` to `20` 

The learning rate used for training the model from scratch was settled to : 
```
__C.TRAIN.LR_INIT             = 1e-5
__C.TRAIN.LR_END              = 1e-6
``` 

I'm restoring as follow :  
```
$ python3 train.py --model="yolov4" --weights="./checkpoints/yolov4"
```

Any advices ?

**Comment by MI-LA01** [comment]
Answering to other, the current implementation don't `manage` the checkpoints. I finally fixed it by implementing something with : 

```
ckpt        = tf.train.Checkpoint(step=tf.Variable(1, trainable=False, dtype=tf.int64), optimizer=optimizer, net=model)
    manager     = tf.train.CheckpointManager(ckpt, './checkpoints', max_to_keep=10)
```
And then used ```ckpt.restore()``` 

This imply to get rid of ```global_steps``` and replace with the ```ckpt.step```
